lavalink/websocket.py

```diff
@@ -0,0 +1,128 @@
+import asyncio
+import logging
+import aiohttp
+from .stats import Stats
+from .events import TrackEndEvent, TrackExceptionEvent, TrackStuckEvent
+
+log = logging.getLogger('lavalink')
+
+
+class WebSocket:
+    def __init__(self, node, host: str, port: int, password: str):
+        self._node = node
+        self._lavalink = self._node._manager._lavalink
+
+        self._session = self._lavalink._session
+        self._ws = None
+        self._message_queue = []
+
+        self._host = host
+        self._port = port
+        self._password = password
+
+        self._shards = self._lavalink._shard_count
+        self._user_id = self._lavalink._user_id
+
+        self._closers = [aiohttp.WSMsgType.close,
+                         aiohttp.WSMsgType.closing,
+                         aiohttp.WSMsgType.closed]
+
+        self._loop = self._lavalink._loop
+        asyncio.ensure_future(self.connect())
+
+    @property
+    def connected(self):
+        """ Returns whether the websocket is connected to Lavalink. """
+        return self._ws is not None and not self._ws.closed
+
+    async def connect(self):
```

Comment:
Just a comment to me: I'll take a look at threading to run ws on a separate loop.

Reply:
Is this not already achieved by using aiohttp in the first place, where it uses a pool of worker threads? Additionally message receiving and event handling is done within a future which I believe assists asynchronous usage?